lib/feature/class-feature.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Automattic\VIP;
+
+class Feature {
+	public static $feature_percentages = array();
+
+	public static $site_id = FILES_CLIENT_SITE_ID;
+
+	public static function is_enabled( $feature ) {
+		return static::is_enabled_by_percentage( $feature );
+	}
+
+	public static function is_enabled_by_percentage( $feature ) {
+		if ( ! isset( static::$feature_percentages[ $feature ] ) ) {
+			return false;
+		}
+
+		$percentage = static::$feature_percentages[ $feature ];
+
+		// Which bucket is the site in - 100 possibilites, one for each percentage
+		$bucket = static::$site_id % 100;
```

Comment:
In order to avoid the same sites being always the first ones where features are enabled, the feature name can be included to compute the bucket. For example:

`$bucket = crc32( $feature . static::$site_id ) % 100;`

I know this makes tests more difficult to understand, so you can leave this as a suggestion to be implemented if this library ends up being very used.

Reply:
I like this idea, it seems worth implementing now.

Reply:
Added the hash in df7d735